lib/checksizeof.cpp

```diff
@@ -373,16 +373,20 @@ void CheckSizeof::suspiciousSizeofCalculation()
     if (!mSettings->severity.isEnabled(Severity::warning) || !mSettings->certainty.isEnabled(Certainty::inconclusive))
         return;
 
-    // TODO: Use AST here. This should be possible as soon as sizeof without brackets is correctly parsed
     for (const Token *tok = mTokenizer->tokens(); tok; tok = tok->next()) {
         if (Token::simpleMatch(tok, "sizeof (")) {
-            const Token* const end = tok->linkAt(1);
-            const Variable* var = end->previous()->variable();
-            if (end->strAt(-1) == "*" || (var && var->isPointer() && !var->isArray())) {
-                if (end->strAt(1) == "/")
-                    divideSizeofError(tok);
-            } else if (Token::simpleMatch(end, ") * sizeof") && end->next()->astOperand1() == tok->next())
-                multiplySizeofError(tok);
+            const Token* lPar = tok->astParent();
+            if (lPar && lPar->str() == "(") {
+                const Token* const end = lPar->link();
+                const Token* const varTok = end->previous();
+                const Variable* var = varTok ? varTok->variable() : nullptr;
+                if ((varTok && varTok->str() == "*") || (var && var->isPointer() && !var->isArray() && !(lPar->astOperand2() && lPar->astOperand2()->str() == "*"))) {
+                    if (lPar->astParent() && lPar->astParent()->str() == "/")
+                        divideSizeofError(tok);
```

Comment:
We used to only write divideSizeof warnings if the sizeof-pointer expression was in LHS. I have the feeling that it was intentional. So I would like to be careful about extending it to checking division RHS also. Could we run some daca@home on this PR and see if there are legitimate cases where this warns?